app/assets/javascripts/Components/submission_file_manager.jsx

```diff
@@ -64,13 +66,34 @@ class SubmissionFileManager extends React.Component {
     }
   }
 
+  handleCreateUrl = (url, url_text) => {
+    this.setState({showURLModal: false});
+    const data_to_upload = {
+      new_url: url,
+      url_text: url_text,
+      path: "/" + (this.state.uploadTarget || ""),
```

Comment:
You're on the right track but this doesn't currently work because `this.state.uploadTarget` is always undefined. You're mimicking the logic of `handleCreateFiles`, which is good, but you also need to check out `handleActionBarAddFileClick` (and possibly other functions).